archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/ast/Input.java

```diff
@@ -692,6 +790,101 @@ public String toString() {
         }
     }
 
+    /**
+     * List of Validation.
+     */
+    public static final class Validations extends Input {
+
+        private final String id;
+        private final String description;
+
+        private Validations(Input.Builder builder) {
+            super(builder);
+            this.id = builder.attribute("id", true).asString();
+            this.description = builder.attribute("description", false).asString();
+        }
+
+        @Override
+        public <A> VisitResult accept(Input.Visitor<A> visitor, A arg) {
+            return visitor.visitValidations(this, arg);
+        }
+
+        @Override
+        public <A> VisitResult acceptAfter(Input.Visitor<A> visitor, A arg) {
+            return visitor.postVisitValidations(this, arg);
+        }
+
+        /**
+         * Get the id.
+         *
+         * @return id
+         */
+        public String id() {
+            return id;
+        }
+
+        /**
+         * Get description.
+         *
+         * @return description
+         */
+        public String description() {
+            return description;
+        }
+    }
+
+    /**
+     * Validation.
+     */
+    public static final class Validation extends Input {
+
+        private final String regex;
+        private final String format;
+
+        private Validation(Input.Builder builder) {
+            super(builder);
+            this.regex = builder.value();
+            this.format = builder.attribute("format", false).asString();
```

Comment:
See the following links:
- https://en.wikipedia.org/wiki/Comparison_of_regular_expression_engines
- https://gist.github.com/CMCDragonkai/6c933f4a7d713ef712145c5eb94a1816

After more thoughts I think we should drop the `format` attribute. We can validate the expressions at archetype build time.

If the AST node invokes `Pattern.compile` in the constructor we validate that the expression is java compatible. 

We can use a java based javascript engine to validate that the expressions are javascript compatible. JDK includes `nashorn` however that's planned for deprecation. We can use `rhino` (`org.mozilla:rhino`) and evaluate some javascript that perform the validation of the expressions.

This logic would be in the archetype maven plugin only, so that we are not adding a new runtime dependency to the engine.